KAFKA-15404: Disable the flaky integration tests.

kamalcph opened the pull request:
The below integration tests are being failed in all the runs, disabling them until we fix them out:

1. kafka.server.DynamicBrokerReconfigurationTest#testThreadPoolResize and 
2. org.apache.kafka.tiered.storage.integration.OffloadAndConsumeFromLeaderTest

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

kamalcph committed:
KAFKA-15404: Fix the flaky DynamicBrokerReconfiguration test.

showuon commented:
Please let me know when this is ready for review. Thanks.

kamalcph commented:
@showuon @divijvaidya @jolshan 

Not able to reproduce this issue locally. I suspect there is a leak in closing the resources during the `Kraft` mode. Tried the below items, but it didn't work:

1. Commented out the `Disabled` tests. Jenkins seems to initialise the test even-though it's disabled.
2. Enabled both `Kraft` and `zk` mode for the `testThreadPoolResize`, it passes locally but fails on Jenkins. 

Could you please take a look and suggest how to fix this?

showuon commented:
Checking

jolshan commented:
@kamalcph so it looks like it isn't a kraft specific issue. Are you saying you also disabled other tests? I believe there is a way to run tests all on their own via Jenkins. I do wonder if the issue is coming from another test.

showuon commented:
@kamalcph , I'm proposing we disable the 
`kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize()`
 and 
`org.apache.kafka.tiered.storage.integration.OffloadAndConsumeFromLeaderTest` first before we fix them. 

WDYT? @satishd @divijvaidya ?

divijvaidya commented:
> WDYT?

Sure, but we can't release 3.6 before fixing them. The thread leaks may be indicating genuine failures and we didn't have a thread leak problem in 3.5.

satishd commented:
Right, these are blocking tests for 3.6.0. These tests need to be fixed. Even if we disable them now, we need to raise a blocker JIRA.

kamalcph committed:
Disabled the below tests to fix the thread leak:

1. kafka.server.DynamicBrokerReconfigurationTest.testThreadPoolResize() and
2. org.apache.kafka.tiered.storage.integration.OffloadAndConsumeFromLeaderTest

showuon reviewed (approved):
LGTM! 
@kamalcph , please
1. open 2 JIRA issues for these 2 disabled tests
2. update the PR title and description accordingly.

Thanks.

kamalcph commented:
> 2\. .OffloadAndConsumeFromLeaderTest

Filed KAFKA-15421 and reopened KAFKA-15399 ticket to track these tests.

jolshan commented:
I'm going to mark the tickets as blockers for 3.6. Otherwise they can get lost.
